src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.netcore.cs

```diff
@@ -2750,10 +2750,17 @@ static T VectorizedSmall8(ref T xRef, ref T yRef, nuint remainder)
 
                 ref T xRef = ref MemoryMarshal.GetReference(x);
                 Vector512<T> resultIndex =
+#if NET9_0_OR_GREATER
+                    sizeof(T) == sizeof(long) ? Vector512<long>.Indices.As<long, T>() :
+                    sizeof(T) == sizeof(int) ? Vector512<int>.Indices.As<int, T>() :
+                    sizeof(T) == sizeof(short) ? Vector512<short>.Indices.As<short, T>() :
+                    Vector512<byte>.Indices.As<byte, T>();
```

Comment:
It's too bad we don't have associated types so we can do something like `Vector<T>.IntegerIndices` which returns a `Vector<TIndex>` or something